src/extension/background-script/actions/nostr/signEventOrPrompt.ts

```diff
@@ -17,15 +23,60 @@ const signEventOrPrompt = async (message: MessageSignEvent) => {
     };
   }
 
+  const enableMessage: MessageAllowanceEnable = {
+    origin: message.origin,
+    args: {
+      host: message.origin.host,
+    },
+    action: "public/webln/enable",
+  };
+
+  await enable(enableMessage);
+
+  const dbPermissions = await db.permissions
+    .where("host")
+    .equalsIgnoreCase(message.origin.host)
+    .toArray();
+
+  const hasPermission = dbPermissions
+    .filter((permission) => permission.enabled)
+    .map(({ method }) => method)
+    .includes("signMessage");
+
   try {
-    const response = await utils.openPrompt<{
-      confirm: boolean;
-    }>({
-      ...message,
-      action: "public/nostr/confirmSignMessage",
-    });
-    if (!response.data.confirm) {
-      throw new Error("User rejected");
+    if (!hasPermission) {
+      const response = await utils.openPrompt<{
+        confirm: boolean;
+        rememberPermission: boolean;
+      }>({
+        ...message,
+        action: "public/nostr/confirmSignMessage",
+      });
+
+      if (!response.data.confirm) {
+        throw new Error("User rejected");
+      }
+
+      if (response.data.rememberPermission) {
+        const matchingAllowance = await db.allowances
+          .where("host")
+          .equalsIgnoreCase(message.origin.host)
+          .first();
+        const allowanceId = matchingAllowance?.id;
+
+        if (!allowanceId) return; // type-guard only
+
+        const permission: DbPermission = {
+          createdAt: Date.now().toString(),
```

Comment:
why do we have those as string again? this sucks because we always need to do useless typecastings actually.

Reply:
Happy to have a look after this PR